src/panels/lovelace/card-features/hui-fan-oscillate-card-feature.ts

```diff
@@ -0,0 +1,157 @@
+    : undefined;
+  if (!stateObj) return false;
+  const domain = computeDomain(stateObj.entity_id);
+  return (
+    domain === "fan" && supportsFeature(stateObj, FanEntityFeature.OSCILLATE)
+  );
+};
+
+@customElement("hui-fan-oscillate-card-feature")
+class HuiFanOscillateCardFeature
+  extends LitElement
+  implements LovelaceCardFeature
+{
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @property({ attribute: false }) public context?: LovelaceCardFeatureContext;
+
+  @state() private _config?: FanOscillateCardFeatureConfig;
+
+  @state() _oscillate?: boolean;
+
+  private get _stateObj() {
+    if (!this.hass || !this.context || !this.context.entity_id) {
+      return undefined;
+    }
+    return this.hass.states[this.context.entity_id!] as FanEntity | undefined;
+  }
+
+  static getStubConfig(): FanOscillateCardFeatureConfig {
+    return {
+      type: "fan-oscillate",
+    };
+  }
+
+  public setConfig(config: FanOscillateCardFeatureConfig): void {
+    if (!config) {
+      throw new Error("Invalid configuration");
+    }
+    this._config = config;
+  }
+
+  protected willUpdate(changedProp: PropertyValues): void {
+    if (
+      (changedProp.has("hass") || changedProp.has("context")) &&
+      this._stateObj
+    ) {
+      const oldHass = changedProp.get("hass") as HomeAssistant | undefined;
+      const oldStateObj = oldHass?.states[this.context!.entity_id!];
+      if (oldStateObj !== this._stateObj) {
+        this._oscillate = this._stateObj.attributes.oscillating as boolean;
+      }
+    }
+  }
+
+  private async _valueChanged(ev: CustomEvent) {
+    const shouldOscillate = (ev.detail as any).value as boolean;
+
+    if (shouldOscillate === this._stateObj!.attributes.oscillating) return;
+
+    const wadOscillating = this._stateObj!.attributes.oscillating as boolean;
+    this._oscillate = shouldOscillate;
+
+    try {
+      await this._updateOscillate(shouldOscillate);
+    } catch (_err) {
+      this._oscillate = wadOscillating;
+    }
+  }
+
+  private async _updateOscillate(oscillate: boolean) {
+    await this.hass!.callService("fan", "oscillate", {
+      entity_id: this._stateObj!.entity_id,
+      oscillating: oscillate,
+    });
+  }
+
+  protected render(): TemplateResult | null {
+    if (
+      !this._config ||
+      !this.hass ||
+      !this.context ||
+      !this._stateObj ||
+      !supportsFanOscilatteCardFeature(this.hass, this.context)
+    ) {
+      return null;
+    }
+
+    const color = stateColorCss(this._stateObj);
+
+    const yesNo = ["no", "yes"] as const;
+    type YesNo = (typeof yesNo)[number];
+    const options = yesNo.map<ControlSelectOption>((oscillating) => ({
+      value: oscillating,
+      label: this.hass!.localize(
+        `ui.common.${oscillating}` as Extract<
+          LocalizeKeys,
+          `ui.common.${YesNo}`
+        >
+      ),
+      path:
+        oscillating === "yes" ? mdiArrowOscillating : mdiArrowOscillatingOff,
+    }));
```

Comment:
This type magic is not necessary
```suggestion
    const options = yesNo.map<ControlSelectOption>((oscillating) => ({
      value: oscillating,
      label: this.hass!.localize(`ui.common.${oscillating}`),
      path:
        oscillating === "yes" ? mdiArrowOscillating : mdiArrowOscillatingOff,
    }));
```

Reply:
That fix the lint error I got when I start the PR, but it breaks the tooltip because `ui.common.${string}` is not one of `LocalizeKeys`. As asked in comment https://github.com/home-assistant/frontend/pull/26519#issuecomment-3182355636, is it acceptable to update `LocalizeKeys` ?